src/main/java/org/aion/wallet/dto/AccountDTO.java

```diff
@@ -12,13 +13,14 @@
     private String balance;  //TODO this has to be BigInteger
     private String name;
     private boolean active;
-    private BufferedImage qrCode;
+    private final BufferedImage qrCode;
 
-    public AccountDTO(final String name, final String publicAddress, final String balance, final String currency) {
+    public AccountDTO(final String name, final String publicAddress, final String balance, final String currency, final BufferedImage qrCode) {
         this.name = name;
         this.publicAddress = TypeConverter.toJsonHex(publicAddress);
         this.balance = balance;
         this.currency = currency;
+        this.qrCode = qrCode;
```

Comment:
what I meant by removing the code duplication was to have something like this in the `AccountDTO`'s constructor:
`this.qrCode = QRCodeUtils.writeQRCode(publicAddress);` 

This way we don't compute the QRCode, which depends only on the address, outside of the account. so there's no room for programmer error.